aep/general/0162/aep.md.j2

```diff
@@ -1,5 +1,396 @@
+revisions as nested collection of the resource. Sometimes, the revisions
+collection can be a top level collection. Exceptions include:
+
+- If resource revisions are meant to have longer lifespan than the parent
+resource. In other words, resource revisions exist after resource deletion.
+
+{% tab proto %}
+
+```proto
+message BookRevision {
+  // The path of the book revision.
+  string path = 1;
+
+  // The snapshot of the book
+  Book snapshot = 2
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // The timestamp that the revision was created.
+  google.protobuf.Timestamp create_time = 3
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+
+  // Other revision IDs that share the same snapshot.
+  repeated string alternate_ids = 4
+    [(google.api.field_behavior) = OUTPUT_ONLY];
+}
+```
+
+- The `message` **must** be annotated as a resource (AIP-123).
+- The `message` name **must** be named `{ResourceType}Revision`.
+
+{% tab oas %}
+
+```json
+{
+    "$schema": "http://json-schema.org/draft-07/schema#",
+    "type": "object",
+    "properties": {
+        "path": {
+            "type": "string"
+        },
+        "snapshot": {
+            "$ref": "#/definitions/Book",
+            "readOnly": true,
+        },
+        "create_time": {
+            "type": "string",
+            "format": "date-time",
+            "readOnly": true,
+        },
+        "alternate_ids": {
+            "type": "array",
+            "items": {
+                "type": "string"
+            },
+            "readOnly": true,
+        }
+    },
+    "required": ["path"],
+}
+```
+
+{% endtabs %}
+
+- The resource revision **must** contain a field with a message type of the
+  parent resource, with a field name of `snapshot`.
+    - The value of `snapshot` **must** be the configuration of the parent
+      at the point in time the revision was created.
+- The resource revision **must** contain a `create_time` field (see [AIP-142][]).
+- The resource revision **may** contain a repeated field `alternate_ids`, which would
+  contain a list of resource IDs that the revision is also known by (e.g. `latest`)
+
+### Creating Revisions
+
+Depending on the resource, different APIs may have different strategies for
+creating a new revision. Specifically examples of strategies include:
+
+- Creating a revision when there is a change to the parent resource
+- Creating a revision when important system state changes
+- Creating a revision when specifically requested
+
+APIs **may** use any of these strategies. APIs **must**
+document their revision creation strategy.
+
+### Resource names for revisions
+
+When referring to specific revision of a resource, the subcollection name
+**must** be named `revisions`. Resource revisions have paths with the format
+`{resource_name}/revisions/{revision_id}`. For example:
+```
+publishers/123/books/les-miserables/revisions/c7cfa2a8
+```
+
+### Server-specified Aliases
```

Comment:
since revisions does allow for user and server specified ids to live side by side with each other - I think we probably want a new AEP on that specifically.

I already have some thoughts on this - we can propose a namespacing scheme as a *should*, and provide that as a way of supporting both user-specified and non-user-specified resources.

Reply:
adding a *should* about that here.